common/src/View/MapDocumentCommandFacade.cpp

```diff
@@ -835,14 +837,23 @@ namespace TrenchBroom {
             for (const auto& entry : faces) {
                 Model::Brush* brush = entry.first;
                 const std::vector<vm::polygon3>& oldPositions = entry.second;
-                const std::vector<vm::polygon3> newPositions = brush->moveFaces(m_worldBounds, oldPositions, delta, pref(Preferences::UVLock));
-                kdl::vec_append(newFacePositions, newPositions);
+                
+                auto result = brush->moveFaces(m_worldBounds, oldPositions, delta, pref(Preferences::UVLock));
+                if (result.is_success()) {
+                    kdl::vec_append(newFacePositions, kdl::get_value(result));
+                } else {
+                    return kdl::visit_error([](GeometryException&& e) {
```

Comment:
`map_error` might be a better name (I'd expect `kdl::visit_error` to return void, and to just do nothing if the result isn't an error)